### PR TITLE
i3-resurrect: init at 1.4.3

### PR DESCRIPTION
**Pull request by nullscm**
###### Motivation for this change
Missing Package i needed for my workflow

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by magnetophon** [comment]
I tried to build this, but didn't get any binaries, afaik:
```
nixpkgs-review pr 97480
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/97480/head:refs/nixpkgs-review/1
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Total 7 (delta 5), reused 7 (delta 5), pack-reused 0
Unpacking objects: 100% (7/7), 1018 bytes | 127.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
   83162ab3b97..23370afcedc  master     -> refs/nixpkgs-review/0
$ git worktree add /home/bart/.cache/nixpkgs-review/pr-97480-1/nixpkgs 23370afcedc46f63ea0dc7328bcb1f40d78887bf
Preparing worktree (detached HEAD 23370afcedc)
Updating files: 100% (22750/22750), done.
HEAD is now at 23370afcedc curlie: 1.3.1 -> 1.5.4
$ nix-env -f /home/bart/.cache/nixpkgs-review/pr-97480-1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit 34c180c16785698e7add57a7b6197f99bea9c8f8
Automatic merge went well; stopped before committing as requested
$ nix-env -f /home/bart/.cache/nixpkgs-review/pr-97480-1/nixpkgs -qaP --xml --out-path --show-trace --meta
Nothing changed
https://github.com/NixOS/nixpkgs/pull/97480
$ nix-shell /home/bart/.cache/nixpkgs-review/pr-97480-1/shell.nix
```

```
i3-resurrect save
i3-resurrect: command not found
```

Am I doing something wrong?

**Comment by magnetophon** [comment]
@vldn-dev I found at least 2 issues:
- You need to add yourself to `nixpkgs/maintainers/maintainer-list.nix`.
- You need to add the following line to `nixpkgs/pkgs/top-level/all-packages.nix`
  `i3-resurrect = python3Packages.callPackage ../applications/window-managers/i3/i3-resurrect.nix { };
`
I suggest to add it after `i3-layout-manager = callPackage ../applications/window-managers/i3/layout-manager.nix { };
`

The reason I said "at least" is because the program is only partially working for me, but that might be either pebcac or an upstream issue.

File: pkgs/applications/window-managers/i3/i3-resurrect.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonApplication, click, i3ipc, psutil, natsort, fetchPypi, xprop, xdotool, importlib-metadata }:
+
+buildPythonApplication rec {
+  pname = "i3-resurrect";
+  version = "1.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+sha256 = "0h181frdwpqfj9agw43qgicdvzv1i7xwky0vs0ksd8h18qxqp4hr";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/i3-resurrect.nix:9]:
```suggestion
    inherit pname version;
    sha256 = "0h181frdwpqfj9agw43qgicdvzv1i7xwky0vs0ksd8h18qxqp4hr";
```

File: pkgs/applications/window-managers/i3/i3-resurrect.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  propagatedBuildInputs = [ click psutil xprop natsort i3ipc xdotool importlib-metadata ];
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/i3-resurrect.nix:13]: Why did you disable the tests?

File: pkgs/applications/window-managers/i3/i3-resurrect.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/JonnyHaystack/i3-resurrect";
+    description = "A simple but flexible solution to saving and restoring i3 workspaces";
+    license = licenses.gpl3;
+    platforms= platforms.linux;
+    maintainers = with maintainers; [ vldn-dev ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/i3-resurrect.nix:23]:
```suggestion
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please also resolve the other review comment. Marking it a draft.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 97480` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by magnetophon** [comment]
In the meantime this one got merged:
https://github.com/NixOS/nixpkgs/pull/104324

**Pull request closed without merging**